index.coffee

```diff
@@ -56,18 +60,21 @@ onProgressHandler = (onProgressPromise, fallbackOnProgress) ->
 	evts = []
 	onProgress = (evt) ->
 		evts.push(evt)
+
+	handlePreviousEvents = (onProgress) ->
+		_.map evts, (evt) ->
+			try
+				onProgress(evt)
+		evts = []
+
 	# Once the onProgressPromise is fulfilled we switch `onProgress` to the real callback,
 	# or the fallback if it fails, and then call it with all the previously received events in order
 	onProgressPromise
 	.then (resolvedOnProgress) ->
-		onProgress = resolvedOnProgress
+		handlePreviousEvents(onProgress = resolvedOnProgress)
 	.catch (e) ->
 		console.warn('error', e)
-		onProgress = fallbackOnProgress
-	.then ->
-		_.map evts, (evt) ->
-			try
-				onProgress(evt)
+		handlePreviousEvents(onProgress = fallbackOnProgress)
```

Comment:
This `handlePreviousEvents` change is not related with 'unknown blob', but it's a bug fix (I haven't seen this bug being reported, but it can be reasoned as follows). The rationale for the pre-existing code was to initially install a provisional progress handler that collects progress events in an array. Then when the final handler is installed, the previous events are fed to it from the array, in order. The bug was the fact that the final handler was installed and _.then,_ in a following iteration of Node's event loop, the previous events were fed. So there was a gap that allowed new events to be fed to the final handler _before_ the old events were fed, which would result in the handler printing events out of order.